Src/HALALMock/Services/Communication/Ethernet/TCP/Socket.cpp

```diff
@@ -138,6 +131,13 @@ Socket::Socket(IPV4 local_ip, uint32_t local_port, IPV4 remote_ip, uint32_t remo
 Socket::Socket(EthernetNode local_node, EthernetNode remote_node):Socket(local_node.ip, local_node.port, remote_node.ip, remote_node.port){}
 
 void Socket::close(){
+	if(is_connecting){
+		is_connecting = false;
+		connection_thread;
```

Comment:
what is this trying to do?

Reply:
now nothing this is the things I didn't nknow how to solve it I'm going to put a destructor